### ISSUE TITLE
Create log sink for Datasette

### ISSUE THREAD

**Issue by bendnorman** [catalyst-cooperative/pudl#1218]
Our datasette instance is hosted using Cloud Run. Cloud Run logs some useful information we would like to save. The Default log sink only retains the logs for 30 days. Create a new log sink that persists all logs to a cloud bucket. 

[Documentation](https://cloud.google.com/logging/docs/export/configure_export_v2) on how to create a new sink.

**Required Permissions:**
You have one of the following IAM roles for the source Cloud project from which you're routing logs:
- *Owner* (roles/owner)
- *Logging Admin* (roles/logging.admin)
- *Logs Configuration Writer* (roles/logging.configWriter)
- permissions to create a destination (BQ, cloud bucket)

@zaneselvans, it might be easiest for you to set all this up instead of granting me permission to create log sinks and destinations (BigQuery and buckets). 

**Questions:**
- What logs do we care about? should we store all of them? Does datasette have documentation about what is logged? There are 32,000 logs from the last month.
- Do we want to store this data in a cloud bucket or BigQuery?

**Comment by bendnorman** [comment]
## Permissions Call baby agenda
* What is in `projects/catalyst-cooperative-pudl/locations/global/buckets/_Default`
* Cloud bucket or Bigquery?
* What logs should we retain?
* Log sink permissions
* Destination permissions

**Comment by bendnorman** [comment]
## Backup BigQuery Data
To periodically backup our BigQuery data we need to create [scheduled table snapshots](https://cloud.google.com/bigquery/docs/table-snapshots-scheduled).

## Default Log Bucket Data

**Comment by zaneselvans** [comment]
Whatever works! It's a one-time transfer.

**Comment by bendnorman** [comment]
Wow that was a doozy! 

I first tried to save the last 30 days of logs in the _Default Logging Bucket to a new BigQuery dataset. Unfortunately, only new logs were saved to the BigQuery dataset and the logs from the last 30 days weren’t copied over. 

Instead, I went over to the log explorer and downloaded all of the logs from the last 30 days in 10k chunks. I appended the json files together and split them into stderr, stdout and requests logs. These were saved as new line delimited json files. 

Once I had 3 json files with the logs, I followed [this documentation](https://cloud.google.com/bigquery/docs/loading-data-cloud-storage-json#appending_to_or_overwriting_a_table_with_json_data) on how to append data to BigQuery tables. I ran into some schema mismatch errors when I used the Auto detect schema feature. I resolved these issues by [creating a json schema file](https://cloud.google.com/bigquery/docs/schemas#creating_a_json_schema_file) from the existing tables in `datasette_logs`. 

Now we have a months worth logs in the `datasette_logs` BigQuery dataset!
